### PR TITLE
Fix double label shift in CohereAsr training loss

### LINKED ISSUE #46894: CohereASR training-loss double-shift bug (same pattern as Moonshine #46784)
### System Info

- `transformers` version: 5.13.0.dev0 (main branch at commit dc5a497c18)
- - Platform: Windows
- - - Python version: 3.x
- - - - Using GPU: N/A (code-level bug, not runtime)

### Who can help?

@eustlb @ebezzam @vasqu

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

aIn `src/transformers/models/cohere_asr/modular_cohere_asr.py`, `CohereAsrForConditionalGeneration.forward()` has a double-shift bug:

1. Labels are shifted right via `shift_tokens_right()` (line 483) to create `decoder_input_ids`
2. 2. Then `self.loss_function()` is called (line 503), which maps to `ForCausalLMLoss`
3. 3. `ForCausalLMLoss` shifts labels again internally (`labels[..., 1:]` in `loss/loss_utils.py` line 63-64)
This means the model trains against `labels[..., 1:]` instead of `labels`.

This is the **exact same bug** that was fixed in Moonshine in PR #46784 (commit d8c235494e). The Moonshine fix replaced `self.loss_function()` with `CrossEntropyLoss()`, but `CohereAsr` — which inherits from Moonshine and overrides `forward` — was not updated.

```python
# Buggy code (line 503 in modular_cohere_asr.py):
loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size)

# Should be:
loss_fct = CrossEntropyLoss()
loss = loss_fct(logits.reshape(-1, self.config.vocab_size), labels.reshape(-1))
```

### Expected behavior

Training loss should be computed directly against `labels` without any additional shifting, since `shift_tokens_right()` already handles the decoder input alignment. This matches the behavior of Whisper, Bart, and the now-fixed Moonshine model. The fix should use `CrossEntropyLoss()` instead of `self.loss_function()`, as done in Moonshine PR #46784.

### PR DESCRIPTION
**Pull request by Diwak4r**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47200)

## What does this PR do?

Fixes a training-loss bug in `CohereAsrForConditionalGeneration` where labels are shifted twice.

`CohereAsrForConditionalGeneration.forward()` (in both `modular_cohere_asr.py` and the generated `modeling_cohere_asr.py`) calls `shift_tokens_right()` on `labels` to build `decoder_input_ids`, then passes the **same** `labels` to `self.loss_function()`. That maps to `ForCausalLMLoss`, which shifts `labels` again internally (`labels[..., 1:]` in `src/transformers/loss/loss_utils.py`). The result is a double shift: the model trains against `labels[..., 1:]` instead of the real `labels`, degrading training.

This is the exact pattern that was fixed for Moonshine in PR #46784. CohereAsr inherits from Moonshine and overrides `forward()` but was never updated.

### Fix
Replace `self.loss_function(logits=..., labels=labels, vocab_size=...)` with a plain `CrossEntropyLoss` against the already-aligned `labels`, matching Whisper, Bart, and the fixed Moonshine:

```python
loss_fct = CrossEntropyLoss()
loss = loss_fct(logits.reshape(-1, self.config.vocab_size), labels.reshape(-1))
```

Applied to the modular source file (source of truth) and mirrored in the auto-generated `modeling_cohere_asr.py`.

## Verification
- Confirmed at the loss level that `ForCausalLMLoss(logits, labels)` produces a value that differs from a direct `CrossEntropyLoss` on `(logits, labels)` (i.e. the double shift is real).
- Added `test_training_loss_no_double_shift` to `tests/models/cohere_asr/test_modeling_cohere_asr.py`, mirroring the Moonshine regression test: for both unpadded and `-100`-padded labels, the reported loss equals an aligned CE and is **not** equal to a double-shifted CE.
- `ruff check` passes on both edited files.

Fixes #46894.

## Before submitting
- [x] This PR fixes a bug (does not add new models/features)
- [x] Tests added/updated

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Duplicate of #46895, blocked for just running your code agent without checking
